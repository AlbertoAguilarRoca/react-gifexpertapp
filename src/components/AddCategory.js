import React, { useState } from 'react';
import PropTypes from 'prop-types';

//recibe por parametro la funcion del hook useState
export const AddCategory = ( {setCategories} ) => {
    //hook useState con la informacion del input 
    const [inputValue, setinputValue] = useState('');

    const handleInputValue = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        //metemos dentro del useState la nueva categoria
        if(inputValue.trim().length > 2) {
            //Metemos el nuevo valor junto con las categorias
            //que ya habia antes
            setCategories(category => [inputValue, ...category ]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className='buscador'>
                <i className='bi bi-plus-circle'></i>
                <input 
                    type='text' 
                    value={ inputValue }
                    onChange = { handleInputValue }
                />
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
