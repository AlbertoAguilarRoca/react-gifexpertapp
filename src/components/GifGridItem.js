import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {

  return (
    <div className='card animate__animated animate__fadeInDown'>
      <div className='card__content'>
        <img src={url} alt={title} />
        <h3>{title}</h3>
      </div>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
