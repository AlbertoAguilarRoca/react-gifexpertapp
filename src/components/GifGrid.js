import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem'

//se recibe la categoria que se va a buscar
export const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    //imprimira las imagenes a traves de un nuevo componente
    //en vez de enviar un parametro, se envia ...img para poder
    //desestructurar dicho componente
    //Cuando el componente se ejecuta varias veces, hay que poner una key unica
    return (
        <>
            <h2 className='animate__animated animate__fadeInDown'>{category}</h2>

            {loading && <p className='animate__animated animate__fadeInDown'>
                Cargando...</p>}

            <div className='card__row'>
                {
                    images.map( img => (
                        <GifGridItem 
                            key = {img.id}
                            {...img} 
                        />
                    ))
                }
            </div>
        </>
    )
}
