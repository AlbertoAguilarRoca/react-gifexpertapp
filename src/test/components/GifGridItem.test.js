import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de <GifGridItem />', () => {
    
    const title = 'TÃ­tulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe mostrar <GifGridItem /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un h3 con el title', () => { 
        const h3 = wrapper.find('h3');
        expect( h3.text().trim() ).toBe(title);
    });

    test('debe de tener la imagen con src y alt', () => { 
        const img = wrapper.find('img');
        //Props divide todo los atributos de un elemento en un objeto
        expect( img.props().src ).toBe(url);
        //Prop permite seleccionar un atributo en concreto
        expect( img.prop('alt') ).toBe(title);
    });

    test('debe contener la clase animate__fadeInDown', () => { 
        const div = wrapper.find('div').at(0);
        //comprueba si tiene la clase
        const hasClass = div.prop('className').includes('animate__fadeInDown');
        expect(hasClass).toBe(true);
     });

});