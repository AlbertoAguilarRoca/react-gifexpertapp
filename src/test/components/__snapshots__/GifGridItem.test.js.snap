// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <div
    className="card__content"
  >
    <img
      alt="Título"
      src="https://localhost/algo.jpg"
    />
    <h3>
      Título
    </h3>
  </div>
</div>
`;
